app/src/main/java/com/example/healthinspector/Fragments/SignupFragment.java

```diff
@@ -0,0 +1,109 @@
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+import java.util.List;
+
+public class SignupFragment extends Fragment {
+    private FragmentSignupBinding binding;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        binding = FragmentSignupBinding.inflate(inflater, container, false);
+        View view = binding.getRoot();
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        binding.btnNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String username = binding.usernameEditText.getText().toString();
+                String password = binding.passwordEditText.getText().toString();
+
+                //input checking
+                if(username.isEmpty()){
+                    Toast.makeText(requireContext(), "Please enter a valid username", Toast.LENGTH_SHORT).show();
+                }
+                else if(password.isEmpty()){
+                    Toast.makeText(requireContext(), "Please enter a valid password", Toast.LENGTH_SHORT).show();
+                }
+                // querying users to check for duplicate username
+                ParseQuery<ParseUser> userQuery = ParseQuery.getQuery(ParseUser.class);
+                userQuery.findInBackground(new FindCallback<ParseUser>() {
+                    @Override
+                    public void done(List<ParseUser> usersFound, ParseException e) {
+                        // check for errors
+                        if (e != null) {
+                            Log.e("", "Issue with getting posts", e);
+                            return;
+                        }
+                        for (ParseUser user : usersFound) {
+                            System.out.println("post");
+                            if(user.getUsername().equals(username)){
+                                Toast.makeText(requireContext(), "Username is already taken", Toast.LENGTH_SHORT).show();
+                            }
+                        }
+                        //username is valid, sign user up
+                        signupUser(username, password);
+                    }
+                });
+            }
+        });
+    }
+
+    private void signupUser(String username, String password){
+        // Creating the new Parse user and navigating the user to their profile
+        ParseUser user = new ParseUser();
+        user.setUsername(username);
+        user.setPassword(password);
+
+        // Invoke signUpInBackground
+        user.signUpInBackground(new SignUpCallback() {
+            public void done(ParseException e) {
+                if(e != null){
+                    return;
```

Comment:
what happens if there is a parse exception while signing up the user? we should show a toast to tell the user to try and give details about what went wrong. We should also log this here so we are able to debug it. We shouldn't just silently return for errors like this.